### PR TITLE
[backend] new wrapper script for docker jailed services

### PR DESCRIPTION
**Pull request by M0ses**
refactored shell based wrapper to perl

### CODE REVIEW

File: src/backend/cservice [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+my $SERVICES_DIR      = $BSConfig::servicetempdir;
+my $OBS_SERVICE_BUNDLE_GEMS_MIRROR_URL = $BSConfig::gems_mirror;
+my $SSH_CMD           = "";
+my $WITH_NET          = 0;
+my $LOGDIR            = "/srv/obs/service/log/";
+my $LOGFILE           = "$LOGDIR/cservice.log";
+open(CPERR, ">&STDERR");
+open(CPOUT, ">&STDOUT");
+open(STDERR, ">>", $LOGFILE) || die "Error stderr: $!";
+open(STDOUT, ">>", $LOGFILE) || die "Error stdout$!";
```
Comment (mlschroe) [inline src/backend/cservice]: Fix error message

Comment (mlschroe) [inline src/backend/cservice]: And please dup stderr to stdout
```
open(STDOUT, ">>", $LOGFILE) || die "Error stdout: $!\n";
open(STDERR, ">&STDOUT") || die "Error stderr: $!\n";
```

File: src/backend/cservice [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+close CPERR;
+close CPOUT;
```
Comment (mlschroe) [inline src/backend/cservice]: Remove that CPOUT/CPERR stuff...

File: src/backend/cservice [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    $COMMAND = "$COMMAND '$arg' '$opt'";
+    printlog("DEBUG COMMAND: $COMMAND");
+  }
+}
+
+die "ERROR: no outdir given!\n" unless $OUTDIR;
+
+my @oparts = split /\//, $OUTDIR;
+pop @oparts;
+my $MOUNTDIR = join'/', @oparts;
```
Comment (mlschroe) [inline src/backend/cservice]: Doesn't perl have some easy way to do a dirname?

File: src/backend/cservice [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+my $RETURN=0;
+create_dir($MOUNTDIR);
+
+print "OUTDIR  : $OUTDIR\n";
+print "MOUNTDIR: $MOUNTDIR\n";
+print Dumper(\@ARGS);
+
+my $tmp = File::Temp::tempnam('', '');
```
Comment (mlschroe) [inline src/backend/cservice]: please use File::Temp::mktemp

File: src/backend/cservice [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+  printlog("DEBUG_DOCKER is set. Skipping cleanup");
+} else {
+  printlog("Starting cleanup");
+  rmdir  "$MOUNTDIR/$INNERSRCDIR";
+  rmdir  "$MOUNTDIR/$INNEROUTDIR";
+  unlink "$MOUNTDIR/${INNERSCRIPT}.command";
+  unlink "$MOUNTDIR/$INNERSCRIPT";
+  rmdir  "$MOUNTDIR$INNERSCRIPTDIR";
+  rmdir  $MOUNTDIR;
+  `docker inspect $CONTAINER_ID > /dev/null 2>&1`;
```
Comment (mlschroe) [inline src/backend/cservice]: system...

**Comment by adrianschroeter** [comment]
you need to  update the spec file (builds failing due to not packaged file)

**Comment by bgeuken** [comment]
@adrianschroeter @M0ses @mlschroe Just noticed that this PR got approved. Can we merge this or is there anything missing?

**Pull request merged**